src/Utils/StrictPlus.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette;
+
+use Nette\Utils\Callback;
+use Nette\Utils\ObjectMixin;
+
+
+/**
+ * Strict class for better experience.
+ * - 'did you mean' hints
+ * - access to undeclared members throws exceptions
+ * - support for @property annotations
+ * - support for calling event handlers stored in $onEvent via onEvent()
+ * - compatible with Nette\Object
+ */
+trait StrictPlus
+{
+
+	/**
+	 * @return mixed
+	 * @throws MemberAccessException
+	 */
+	public function __call($name, $args)
+	{
+		$class = get_class($this);
+		$isProp = ObjectMixin::hasProperty($class, $name);
+
+		if ($name === '') {
+			throw new MemberAccessException("Call to class '$class' method without name.");
+
+		} elseif ($isProp && $this->$name instanceof \Closure) { // closure in property
+			trigger_error("Invoking closure in property via \$obj->$name() is deprecated" . ObjectMixin::getSource(), E_USER_DEPRECATED);
+			return call_user_func_array($this->$name, $args);
+
+		} elseif ($isProp === 'event') { // calling event handlers
```

Comment:
Is "calling event handlers" intentionally handled after "closure in property"? Is the following behavior intentional?

~~~php
$obj->onUpdate = function () { ... };
$obj->onUpdate() // this will call the closure instead of throwing error that property `$onUpdate` must be "array or NULL".
~~~